runtime/caches/redis.ts

```diff
@@ -1,36 +1,28 @@
-import { Redis } from "https://deno.land/x/upstash_redis@v1.22.1/mod.ts";
-
-import { logger, tracer } from "../../observability/otel/config.ts";
 import {
   assertCanBeCached,
   assertNoOptions,
-  withCacheNamespace,
+  requestURLSHA1,
 } from "./common.ts";
+import { createClient } from "npm:redis@^4.5";
 
-const redisUrl = Deno.env.get("UPSTASH_REDIS_REST_URL");
-const redisToken = Deno.env.get("UPSTASH_REDIS_REST_TOKEN");
-
-export const redis = redisUrl && redisToken
-  ? new Redis({
-    url: redisUrl,
-    token: redisToken,
-    enableTelemetry: true,
-  })
-  : null;
-
-interface ResponseMetadata {
-  body: string;
-  status: number;
-  headers: [string, string][];
-}
+const REDIS_URL = Deno.env.get("REDIS_URL") || "redis://localhost:6379/0";
 
-function base64encode(str: string): string {
-  return btoa(encodeURIComponent(str));
+function serialize(response: Response): Promise<string> {
```

Comment:
an interesting perf test to do is to use a string concatenated with special characters and then split it when read from redis, wondering if JSON operations makes it too slow